composer/core/precision.py

```diff
@@ -54,5 +61,17 @@ def get_precision_context(precision: Union[str, Precision]) -> Generator[None, N
         else:
             os.environ['XLA_USE_BF16'] = '1'
             yield
+    elif precision == Precision.AMP_FP8:
+        if te_installed and torch.cuda.get_device_capability()[0] > 8:
+            from transformer_engine.common.recipe import DelayedScaling, Format
+            fp8_format = Format.HYBRID  # E4M3 during forward pass, E5M2 during backward pass
+            fp8_recipe = DelayedScaling(fp8_format=fp8_format, amax_history_len=16, amax_compute_algo='max')
+            with te.fp8_autocast(enabled=True, fp8_recipe=fp8_recipe):
+                yield
+        else:
+            if te_installed:
+                raise ValueError(f'AMP_FP8 precision is used but current device does not support it.')
```

Comment:
suggest `RuntimeError`

Reply:
I think `ValueError` is correct -- we generally use it for invalid parameters to Trainer. For example, see the else case here